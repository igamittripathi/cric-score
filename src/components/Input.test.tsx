import * as React from 'react';
import { Input } from './Input';
import { shallow } from "enzyme";
import { createShallow } from '@material-ui/core/test-utils';
import { act } from 'react-dom/test-utils';

describe("Input", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Input />);
    });

    test("Input should render properly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("allows us to set props", () => {
        expect(wrapper.props().value).toBeUndefined();
        wrapper.setProps({ value: 'india' });
        expect(wrapper.props().value).toEqual('india');
    });

    test('allows us to onChange events', () => {
        const onChange = jest.fn();
        wrapper = createShallow()(<div> <Input error={false} required={true} onChange={onChange}/></div>);
        act(() => {
            wrapper.find(Input).simulate('change', { target: { value: 'James' } });
        });
        wrapper.update();
        expect(wrapper.find(Input).at(0).props().error).toBeFalsy();
    });

});