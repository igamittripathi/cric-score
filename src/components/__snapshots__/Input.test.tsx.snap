// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input should render properly 1`] = `
<WithStyles(ForwardRef(InputBase))
  classes={
    Object {
      "input": "WithStyles(ForwardRef(InputBase))-input-2",
      "root": "WithStyles(ForwardRef(InputBase))-root-1",
    }
  }
/>
`;
